lesson3/hometask.rb

```diff
@@ -0,0 +1,30 @@
+
+module MyEach
+
+  def my_each
+    return self.to_enum unless block_given?
```

Comment:
@oefimenko, nice catch, you did more than I expected.

But you missed an argument in that line. If you leave it blank [to use option by default](https://ruby-doc.org/core-2.3.1/Object.html#method-i-to_enum), it would call general :each, not our custom one. 

It's easy to check. Let's add `puts 'inside my_each'` as second line in `#my_each` (just below `return self.to_enum unless block_given?`). In your case calling something like:
```ruby
[1, 2, 3, 4].my_each.map { |i| i }
```
returns `[1, 2, 3, 4]`, but prints nothing.

So, there are no `inside my_each` in output. But if you change a first line of the method to 
```ruby
return self.to_enum(:my_each) unless block_given?
```
it returns the same `[1, 2, 3, 4]`, but also you will see on output:

> inside my_each

It means that your postponed enumerator will be calling in case of updated version of the script instead of calling default `#each`.